src/utilities/effects.ts

```diff
@@ -3714,10 +3726,33 @@ const effects = {
       if (!queryPermissions.UPDATE_ACTIVITY_PRESET(user, updatedActivityPreset)) {
         throwPermissionError('update this activity preset');
       }
+      if (
```

Comment:
Same as the above for here (but not for the same reason), if we already have a way of passing in the final arguments to the preset creation, I think we could just reuse that for here rather than making the extra query.